astropy/units/format/generic.py

```diff
@@ -594,11 +594,6 @@ def _do_parse(cls, s, debug=False):
                 else:
                     raise ValueError(f"Syntax error parsing unit '{s}'")
 
-    @classmethod
```

Comment:
This is really the main change. Since FITS and VOUNIT inherit from `generic`, this means that they had the same sorting -- undoing the more logical sorting in their classes. Indeed, the sorting that was done in FITS/VOUNIT just followed what `CompositeUnit` itself does anyway, so in that sense for those formats this change makes them behave as intended...